### PR TITLE
Standardise spelling and casing of homeserver, identity server, and integration manager 

### PR DESCRIPTION
**Pull request by psrpinto**
This PR standardises the spelling and casing of the following terms, across the codebase, as per vector-im/element-web#10569:

- homeserver
- identity server
- integration manager

While the issue mentions only user-visible text, this PR changes all instances of the terms, regardless of whether they're user-visible or not. I think there's value in standardising these terms even in non-user-visible text, so I went ahead and applied the changes to all text.

For reference, here's the script I used to find candidates for replacement: https://gist.github.com/psrpinto/b3787bae212d5d99649b517e2efd4dce

### CODE REVIEW

**Comment by t3chguy** [comment]
Only the en_EN file can be modified here, other files must be left for Weblate exclusively https://github.com/vector-im/element-web/blob/develop/docs/translating.md

File: CHANGELOG.md [hunk 1]
```diff
@@ -4933,7 +4933,7 @@ All Changes
    [\#3869](https://github.com/matrix-org/matrix-react-sdk/pull/3869)
  * Move feature flag check for new session toast
    [\#3865](https://github.com/matrix-org/matrix-react-sdk/pull/3865)
- * Catch exception in checkTerms if no ID server
+ * Catch exception in checkTerms if no identity server
    [\#3863](https://github.com/matrix-org/matrix-react-sdk/pull/3863)
  * Catch exception if passphrase dialog cancelled
```
Comment (SimonBrandner) [inline CHANGELOG.md:4938]: Also please don't modify the changelogs

**Comment by psrpinto** [comment]
> Only the en_EN file can be modified here, other files must be left for Weblate exclusively https://github.com/vector-im/element-web/blob/develop/docs/translating.md

That makes sense, I was actually wondering how translations work, should probably have checked that before making changes. Thanks for the link!

I'm on this.

**Comment by psrpinto** [comment]
I think this should be ready for review now. 

On the subject of not losing translated strings when a translations key changes, there was some [discussion on element-tranlations:matrix.org](https://matrix.to/#/!yvhoprZiKmhjxKfNHa:matrix.org/$OwxdNcBMpNqckXbE6BkuUy2crp6FkJU5DzIdNe-OKV0?via=matrix.org&via=thewired.ch&via=tchncs.de). Finally, from what I understand, there's no other way to do this than to re-translate the string through Weblate. Once this is merged, I will try to recover (some of) those translations through weblate.

**Review by SimonBrandner — APPROVED** [review]
LGTM, thanks for working on this! Though ultimately the review here is up to repo owners

**Review by gsouquet — APPROVED** [review]

**Pull request merged**